django/db/models/indexes.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import unicode_literals
+
+import hashlib
+
+from django.utils.encoding import force_bytes
+
+__all__ = ['Index']
+
+# The max length of the names of the indexes (restricted to 30 due to Oracle)
+MAX_NAME_LENGTH = 30
+
+
+class Index(object):
+
+    index_type = 'idx'
+
+    def __init__(self, fields=[], name=None):
+        if not fields:
+            raise ValueError('At least one field is required to define an index.')
+        self.fields = fields
+        self._name = name or ''
+        if self._name:
+            self.normalise_name()
+            if len(self._name) > MAX_NAME_LENGTH:
+                raise ValueError("Index names cannot be longer than %s characters." % MAX_NAME_LENGTH)
+
+    @property
+    def name(self):
+        if not self._name:
+            self._name = self.get_name()
+            self.normalise_name()
+        return self._name
+
+    def normalise_name(self):
+        # Name shouldn't start with an underscore (Oracle hates this), so prepend D if we need to
+        if self._name[0] == "_":
+            self._name = 'D%s' % self._name[1:]
+        # It can't start with a number on Oracle, so prepend D if we need to
+        elif self._name[0].isdigit():
+            self._name = 'D%s' % self._name[1:]
+
+    def create_sql(self, schema_editor):
+        columns = [field for field in self.fields]
+        fields = [self.model._meta.get_field(field) for field in self.fields]
+        if len(fields) == 1 and fields[0].db_tablespace:
```

Comment:
I guess none of these lines aren't tested since I don't see any new tests that use `db_tablespace`.